pkg/status/workstatus.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+type workStatus struct {
+	wecName                string
+	sourceObjectIdentifier util.ObjectIdentifier
+	status                 *runtime.RawExtension
```

Comment:
Maybe this field type should be `map[string]interface{}`? This will prevent unnecessary conversion.

Reply:
I prefer keeping this struct identical to the workstatus. Overall, someone will have to do this conversion since this is what WorkStatus objects contain.

Reply:
The caller code gets the `map[string]interface{}` here https://github.com/vMaroon/kubestellar/pull/11/files#diff-b454b6b87e03b38df5fd12bcd7217e0dd525d4278d5b02c5c4de2c5e2a1f83c4R85, and the consuming code converts to `map[string]interface{}` here https://github.com/kubestellar/kubestellar/pull/2174/files#diff-faa0ed061f946759218486d1b63d7652398765a0e05099b16501ce467c6285d5R89
So if we change the `workStatus.status` type to `map[string]interface{}` there will be no conversion at all.

Reply:
Ah sorry I did not see that yet, will review (and change this field)